### PR TITLE
fix (RedisRegistry): avoid overwriting cache with the same key during load 

### PR DESCRIPTION
**Pull request by ab93**
- fix cache key overwrite issue which caused a stale model to persist for a long time
- Save sourcetype in artifact data extras to be used for other logic
- Improve docs

### CODE REVIEW

**Review by s0nicboOm — APPROVED** [review]

File: numalogic/registry/redis_registry.py [hunk 1]
```diff
@@ -221,24 +249,28 @@ def save(
...
+        ------
+            RedisRegistryError: If there is any RedisError while saving the artifact.
         """
         key = self.construct_key(skeys, dkeys)
         latest_key = self.__construct_latest_key(key)
         version = 0
         try:
             if self.client.exists(latest_key):
-                _LOGGER.debug("latest key exists for the model")
+                _LOGGER.debug("Latest key exists for the model")
```
Comment (vigith) [inline numalogic/registry/redis_registry.py:263]: print the latest_key too?

File: numalogic/registry/redis_registry.py [hunk 2]
```diff
@@ -123,20 +124,38 @@ def __get_artifact_data(
...
+
+        Raises
+        ------
+            ModelKeyNotFound: If the model key is not found in the registry.
+        """
         cached_artifact = self._load_from_cache(key)
         if cached_artifact:
             _LOGGER.debug("Found cached artifact for key: %s", key)
-            return cached_artifact
+            return cached_artifact, True
```
Comment (vigith) [inline numalogic/registry/redis_registry.py:149]: can you make the return an explicit tuple?

**Review by vigith — APPROVED** [review]
with few comments

File: numalogic/registry/redis_registry.py (hunk 2 above)
Comment (ab93) [reply numalogic/registry/redis_registry.py:149]: Pylint complains about redundant parenthesis

**Pull request merged**